File: Kap15/C#/Listing06/ByteReader.cs
class ByteReader {
    public static void printBytes(String fName, int noOfBytes) {
        byte[] myBuffer = new byte[noOfBytes];
        int readBytes = 0;
        using (FileStream myStream = File.Open(fName, FileMode.Open)) {
            readBytes = myStream.Read(myBuffer, 0, noOfBytes); 
            for (int i = 0; i < readBytes; i++) {
                Console.WriteLine(myBuffer[i].ToString("X"));
            }
        }
    }
}

File: Kap15/Java/Listing06/ByteReader.java
package Kap15.Java.Listing06;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;

public class ByteReader {
    public static void main(String[] args) {
        try {
          ByteReader.printBytes("C:\\temp\\MyBitmap.bmp", 100);
        } catch (IOException e) {
          e.printStackTrace();
        }
    }
    
    public static void printBytes(String fName, int noOfBytes) throws IOException {
      File myFile = new File(fName);
      byte[] myBuffer = new byte[noOfBytes];
      int readBytes = 0;
      try (FileInputStream fis = new FileInputStream(myFile)){
        readBytes = fis.read(myBuffer, 0, noOfBytes);
      }
      for (int i = 0; i < readBytes; i++) {
        System.out.println(Integer.toHexString(myBuffer[i]));
      }
    }
    
    
}